design/pipelines.go

```diff
@@ -0,0 +1,29 @@
+package design
+
+import (
+	d "github.com/goadesign/goa/design"
+	a "github.com/goadesign/goa/design/apidsl"
+)
+
+var _ = a.Resource("pipelines", func() {
+	a.BasePath("/pipelines")
+
+	// An auth token is required to call the auth API to get an OpenShift auth token.
+	a.Security("jwt")
+
+	a.Action("delete", func() {
+		a.Routing(
+			a.DELETE(""),
+		)
+		a.Description("Delete pipelines")
+		a.Params(func() {
+			a.Param("space", d.String, "Name of the space")
+			a.Required("space")
+		})
```

Comment:
@hrishin as discussed, we need to change endpoint
From: `/api/pipelines?space=<spaceLable>`
To: `/api/spaces/:spaceID/pipelines`

Two changes:
1. spaceID instead of spaceLable
2. spaceID part of path instead of query

Reply:
@alexeykazakov when we want to `deleteAll` for given `spaceID`, we can go ahead with above rest_api_path.
Got reference from - https://github.com/fabric8-services/fabric8-auth/blob/master/design/collaborators.go#L69

Reply:
`/api/spaces/:spaceID/pipelines` looks good.

Reply:
@kwk the final endpoint should be `/api/spaces/:spaceID/pipelines`.
The reference, I have mentioned to show `deleteAll` endpoint example and not bothering about 
 `BasePath()`.